openedx/edx-enterprise
# feat: remove content transmission audits without a catalog uuid

## Description
There are currently content transmission audits that have no catalog UUID’s associated with them. As the system currently relies on these catalog uuids to update and read audits, these records have become effectively orphaned and useless. As such we would like to clean them up and remove them from the table.

## Solution
Added a command to remove content transmission audit records without a catalog UUID.

## Test Plan
1. Checkout branch `knguyen2/ENT-7268`
2. Visit this URL: http://localhost:18000/admin/integrated_channel/contentmetadataitemtransmission/ and click on the button `ADD CONTENT METADATA ITEM TRANSMISSION`. You will need to leave the `Enterprise customer catalog uuid` field empty.
3. In your terminal run the following commands:
```
     $ ssh devstack
     $ cd edx-repos/devstack
     $ make lms-shell
     $ ./manage.py lms remove_null_catalog_transmission_audits
```
4. Go back to the URL: http://localhost:18000/admin/integrated_channel/contentmetadataitemtransmission/ and ensure that the entry you created has been removed. 

[JIRA Ticket](https://2u-internal.atlassian.net/browse/ENT-7268)

**Merge checklist:**
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.

**Post merge:**
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.
